addons/web/static/src/views/basic_relational_model.js

```diff
@@ -369,7 +369,10 @@ export class Record extends DataPoint {
                     continue;
                 case "one2many":
                 case "many2many":
-                    if (!(await this.checkX2ManyValidity(fieldName))) {
+                    if (
+                        (this.isRequired(fieldName) && this.data[fieldName].count === 0) ||
```

Comment:
I don't think it's correct (or at least, I don't think it's how it worked before). For many2manyTags yes, but not for classical x2manys. What's the scenario to reproduce? (Note: it's always a good idea to write a scenario in the commit message).

Also, if we need to do that, I'd put the logic in `checkX2ManyValidity`